mewarpx/mewarpx/mwxrun.py

```diff
@@ -103,27 +103,40 @@ def init_timestep(self, V_anode, DT=None, CFL_factor=None, V_grid=5):
         return self.dt
 
 
-    def init_run(self):
+    def init_run(self, restart=False, force_restart=False, checkpoint_prefix="chkpoint"):
         if self.initialized:
             raise RuntimeError(
                 "Attempted to initialize the mwxrun class multiple times.")
-
-        self.simulation.initialize_inputs()
-        self.simulation.initialize_warpx()
-
-        self.me = _libwarpx.libwarpx.warpx_getMyProc()
-        self.n_procs = _libwarpx.libwarpx.warpx_getNProcs()
-
-        # loop over all species to build their pid_dict's
-        for species in self.simulation.species:
-            species.init_pid_dict()
-
-        # A level is needed for many things like level number. For now I'm
-        # statically setting the default level here. I'm not sure of pitfalls
-        # or how to handle it more generally yet.
-        self.lev = _libwarpx.libwarpx.warpx_finestLevel()
-
-        self.initialized = True
+        try:
+            if restart:
+                init_restart_util.run_restart(checkpoint_prefix, force_restart)
```

Comment:
We should follow the strategy used in `metools` here, where by default a run would restart if there are restart files available but there aren't any it will still happily start from the beginning. However, if `restart=False`, the run should always start from the beginning and if `restart=True` it should throw an error if it cannot restart. To achieve this, the default value for `restart` can be `None` and the logic below should (hopefully work):
```suggestion
            if restart != False:
                init_restart_util.run_restart(checkpoint_prefix, restart)
```

Reply:
Doesn't this remove the possibility of doing a fresh run if no checkpoint files are found? I'll tag @peterscherpelz because he mentioned this to me as something to include calling it 'trying' a restart. Whereas 'forcing' a restart is the behavior you suggest above for `restart=True` - throwing and error and exiting if there is a problem with restarting.